### PR TITLE
[GH-3983] upgrade_mattermost: fixing `find` failure to delete non-empty dir

### LINKED ISSUE #3983: Upgrade_mattermost.sh `find` command fails to delete non-empty directories in opt/mattermost dir
I am having troubles deploying new mattermost version using the upgrade script.
This issue is not a big deal but would be great to have it fixed.

In cases directories in path `/opt/mattermost` are not empty the folowwing: 
```
# Clean up mattermost directory
find "${mattermostdir}" -mindepth 1 -maxdepth 1 -not \( -path "${mattermostdir}/config" -o -path "${mattermostdir}/logs" -o -path "${mattermostdir}/plugins" -o -path "${mattermostdir}/data" \) -delete;
```
will fail to delete the no-empty directories and then avorting script because of the `set -e` enabled.

If you don't mind, I would suggest we replace `-delete` by the `-exec rm -rf {} \;` argument, therefore forcing the delete.

### PR DESCRIPTION
**Pull request by Ant0wan**
#### Summary

*Changes:*
- Replace the find `-delete` argument by `-exec rm -rf {} \;` in order to have it to delete non empty directories located in `/opt/mattermost`. It enables script to delete targeted directories and continue execution.

#### Ticket Link

Closes #3983

### CODE REVIEW

**Comment by justinegeffen** [comment]
Tagging @JtheBAB here as this is related to https://github.com/mattermost/docs/pull/3980. :)

File: source/administration/upgrade-script.rst [hunk 1]
```diff
@@ -41,101 +42,117 @@ Copy and save the following script to your Mattermost server as ``upgrade_matter
...
+   			exit 1
+   		fi
+   	fi
+
+   	echo "[+] The Mattermost update package has been downloaded with successful"
+   }
+
+   # Check previous download
+   if [ -e "${downloaddir}/mattermost-upgrade.tar.gz" ]; then
+   	read -r -p "[?] A previous download exists. Do you want to replace it by a new one? [Y/n] " input
```
Comment (justinegeffen) [inline source/administration/upgrade-script.rst:127]:
```suggestion
   	read -r -p "[?] A previous download exists. Do you want to replace it by a new one? [Y/n " input
```

File: source/administration/upgrade_mattermost.sh [hunk 2]
```diff
@@ -30,96 +31,117 @@ backupdatabase=0
...
-    case "$input" in
-        [yY])
-            echo "[+] Starting the update process of Mattermost..."
-        ;;
-        *)
-            echo "[-] Please create a backup and start again"
-            exit 1
-        ;;
-    esac
+	read -r -p "[?] Do you have a current backup of the Mattermost database? [Y/n] " input
```
Comment (justinegeffen) [inline source/administration/upgrade_mattermost.sh:62]:
```suggestion
	read -r -p "[?] Do you have a current backup of the Mattermost database? [Y/n]" input
```

File: source/administration/upgrade_mattermost.sh [hunk 3]
```diff
@@ -30,96 +31,117 @@ backupdatabase=0
...
+			exit 1
+		fi
+	fi
+
+	echo "[+] The Mattermost update package has been downloaded with successful"
+}
+
+# Check previous download
+if [ -e "${downloaddir}/mattermost-upgrade.tar.gz" ]; then
+	read -r -p "[?] A previous download exists. Do you want to replace it by a new one? [Y/n] " input
```
Comment (justinegeffen) [inline source/administration/upgrade_mattermost.sh:116]:
```suggestion
	read -r -p "[?] A previous download exists. Do you want to replace it by a new one? [Y/n]" input
```

File: source/administration/upgrade-script.rst [hunk 4]
```diff
@@ -41,101 +42,117 @@ Copy and save the following script to your Mattermost server as ``upgrade_matter
...
+   		*)
+   			echo "[-] Please create a backup and start again"
+   			exit 1
+   			;;
+   	esac
    fi
 
-   # Check if Mattermost exists in the path provided above
-
+   # Check if mattermost exists in the path provided above
```
Comment (justinegeffen) [inline source/administration/upgrade-script.rst:86]:
```suggestion
   # Check if Mattermost exists in the path provided above
```

File: source/administration/upgrade-script.rst [hunk 5]
```diff
@@ -146,20 +163,26 @@ Copy and save the following script to your Mattermost server as ``upgrade_matter
    GROUP="$(stat -c '%G' ${mattermostdir}/bin/mattermost)"
    chown -hR "$USER":"$GROUP" "${downloaddir}/mattermost-upgrade/"
 
-   # Clean up Mattermost directory
 
-  find "${mattermostdir}" -mindepth 1 -maxdepth 1 -not \( -path "${mattermostdir}/config" -o -path "${mattermostdir}/logs" -o -path "${mattermostdir}/plugins" -o     -path "${mattermostdir}/data" \) -delete
+   # Clean up mattermost directory
```
Comment (justinegeffen) [inline source/administration/upgrade-script.rst:167]:
```suggestion
   # Clean up Mattermost directory
```

File: source/administration/upgrade_mattermost.sh [hunk 6]
```diff
@@ -30,96 +31,117 @@ backupdatabase=0
...
+			echo "[+] Starting the update process of Mattermost..."
+			;;
+		*)
+			echo "[-] Please create a backup and start again"
+			exit 1
+			;;
+	esac
 fi
 
 # Check if mattermost exists in the path provided above
```
Comment (justinegeffen) [inline source/administration/upgrade_mattermost.sh:75]:
```suggestion
# Check if Mattermost exists in the path provided above
```

File: source/administration/upgrade_mattermost.sh [hunk 7]
```diff
@@ -130,18 +152,26 @@ USER="$(stat -c '%U' ${mattermostdir}/bin/mattermost)"
 GROUP="$(stat -c '%G' ${mattermostdir}/bin/mattermost)"
 chown -hR "$USER":"$GROUP" "${downloaddir}/mattermost-upgrade/"
 
+
 # Clean up mattermost directory
```
Comment (justinegeffen) [inline source/administration/upgrade_mattermost.sh:156]:
```suggestion
# Clean up Mattermost directory
```

**Comment by svelle** [comment]
Sorry for not getting back on this earlier, but I don't think I'm the right person to review this as I'm really unfamiliar with the upgrade script.
@justinegeffen do you know who opened the original request that added this to the docs?

**Comment by justinegeffen** [comment]
Thanks, @svelle! No worries. The doc was written by @JtheBAB and added to the community-documentation section. As it is community-managed I think it's OK to merge but just wanted some fresh eyes on it before I do. :)

**Comment by JtheBAB** [comment]
LGTM 👍

**Comment by Ant0wan** [comment]
Hi ! Thanks a lot for your review. 😀

**Review by justinegeffen — APPROVED** [review]

**Pull request merged**